src/mame/includes/lynx.h

```diff
@@ -29,98 +28,188 @@ class lynx_state : public driver_device
 public:
 	lynx_state(const machine_config &mconfig, device_type type, const char *tag) :
 		driver_device(mconfig, type, tag),
-		m_mem_0000(*this, "mem_0000"),
-		m_mem_fc00(*this, "mem_fc00"),
-		m_mem_fd00(*this, "mem_fd00"),
-		m_mem_fe00(*this, "mem_fe00"),
-		m_mem_fffa(*this, "mem_fffa"),
+		m_dram(*this, "dram"),
 		m_maincpu(*this, "maincpu"),
 		m_sound(*this, "custom"),
 		m_cart(*this, "cartslot"),
 		m_palette(*this, "palette"),
 		m_screen(*this, "screen"),
-		m_bank_fc00(*this, "bank_fc00"),
-		m_bank_fd00(*this, "bank_fd00"),
-		m_bank_fe00(*this, "bank_fe00"),
-		m_bank_fffa(*this, "bank_fffa")
+		m_suzy_view(*this, "suzy_view"),
+		m_mikey_view(*this, "mikey_view"),
+		m_rom_view(*this, "rom_view"),
+		m_vector_view(*this, "vector_view"),
+		m_joy_io(*this, "JOY"),
+		m_pause_io(*this, "PAUSE")
 	{ }
 
 	void lynx(machine_config &config);
 
+protected:
+	virtual void machine_start() override;
+	virtual void machine_reset() override;
+	virtual void video_start() override;
+
+	virtual void device_timer(emu_timer &timer, device_timer_id id, int param, void *ptr) override;
+	virtual void device_post_load() override;
+
 private:
 	struct BLITTER
 	{
+		// 0xfc80 SPRCTL0 Sprite control bits 0
+		const inline u8     BPP()            { return BIT(spr_ctl0, 6, 2); } // Bit per pixel - 1
+		const inline bool   HFLIP()          { return BIT(spr_ctl0, 5); }    // Horizontal flip
+		const inline bool   VFLIP()          { return BIT(spr_ctl0, 4); }    // Vertical flip
+		const inline u8     SPRITE_TYPE()    { return BIT(spr_ctl0, 0, 3); } // Sprite type
+
+		// 0xfc81 SPRCTL1 Sprite control bits 1
+		const inline bool   RLE()            { return BIT(spr_ctl1, 7); }    // Normal(1) or RLE(0) sprite
+		//const inline bool ALGO3()          { return BIT(spr_ctl1, 6); }    // Size algorithm, Adder(0, Algo4), Shifter(1, algo3) (not implemented)
+		const inline u8     RELOAD_SCALE()   { return BIT(spr_ctl1, 4, 2); } // Reload scale factors
+		const inline bool   REUSE_PALETTE()  { return BIT(spr_ctl1, 3); }    // Reload(0) or Reuse(1) palette
+		const inline bool   SKIP_SPRITE()    { return BIT(spr_ctl1, 2); }    // Skip this sprite
+		const inline u8     DRAW_ORIGIN()    { return BIT(spr_ctl1, 0, 2); } // Start draw origin
+		//const inline bool DRAW_UP()        { return BIT(spr_ctl1, 1); }    // Start drawing up(1) or down(0)
+		//const inline bool DRAW_LEFT()      { return BIT(spr_ctl1, 0); }    // Start drawing left(1) or right(0)
+
+		// 0xfc82 SPRCOLL Sprite collision number
+		const inline bool   SPRITE_COLLEN()  { return BIT(spr_coll, 5); }    // Sprite collide(0) or Don't collide(1) flag
+		const inline u8     SPRITE_COLNUM()  { return BIT(spr_coll, 0, 4); } // Sprite collision number
+
+		const inline bool   SPRITE_COLLIDE() { return sprite_collide && (!(no_collide)); } // collision enabled?
```

Comment:
These member functions have the same issues as the ones in the sound device registers structure..